### PR TITLE
Add Tasks Samples and Tests

### PR DESCRIPTION
**Pull request by averikitsch**

### CODE REVIEW

**Comment by averikitsch** [comment]
@ace-n review?

File: samples/package.json [hunk 1]
```diff
@@ -4,17 +4,20 @@
...
   },
   "dependencies": {
-    "@google-cloud/tasks": "0.1.0"
+    "@google-cloud/tasks": "0.1.0",
+    "yargs": "10.0.3"
   },
   "devDependencies": {
-    "@google-cloud/nodejs-repo-tools": "^2.3.0"
+    "@google-cloud/nodejs-repo-tools": "^2.3.0",
+    "ava": "0.23.0"
```
Comment (JustinBeckwith) [inline samples/package.json]: Pretty sure ava is on `0.25.0` - can you make sure we're using the latest and greatest here?  I like to `npm install -g npm-upgrade`, then run `npm-upgrade` in the directory with the package.json.

File: samples/tasks.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    parent: parent,
+    task: task
+  };
+
+  // Send create task request.
+  client.createTask(request).then(response => {
+    const task = response[0].name;
+    console.log(`Created task ${task}`);
+  }).catch(err => {
+    console.log(`Error in createTask: ${err.message || err}`);
```
Comment (JustinBeckwith) [inline samples/tasks.js]: I'm being a little pedantic, but can we use `console.error` here?

File: synth.py [hunk 3]
```diff
@@ -1,3 +1,19 @@
+# Copyright 2018 Google LLC
```
Comment (JustinBeckwith) [inline synth.py]: Thank you :)

File: samples/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+Sample command-line program for interacting with the Google Cloud Tasks API
+using pull queues.
+
+Pull queues let you add tasks to a queue, then programatically remove and
+interact with them. Tasks can be added or processed in any environment,
+such as on Google App Engine or Google Compute Engine.
+
+`tasks.js` is a simple command-line program to demonstrate listing queues,
+ creating tasks, and pulling and acknowledging tasks.
```
Comment (grayside) [inline samples/README.md]: what does "acknowledging" a task mean?

File: samples/tasks.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    {},
+    (opts) => pullTask(opts.project, opts.location, opts.queue)
+  )
+  .command(
+    `acknowledge <task>`,
+    `Acknowledge a task.`,
+    {},
+    (opts) => acknowledgeTask(JSON.parse(opts.task))
+  )
+  .example(`node $0 create my-project-id us-central1 my-queue`)
```
Comment (grayside) [inline samples/tasks.js]: Should these examples use curly braces? `${0}`

Comment (averikitsch) [reply samples/tasks.js]: I don't believe so. Other node samples just use $0.

File: samples/README.md (hunk 4 above)
Comment (averikitsch) [reply samples/README.md]: The worker must acknowledge the task after they have performed the work associated with the task. Should the readme include this information?

Comment (grayside) [reply samples/README.md]: That's what I was getting at, but on second thought I'm not sure what the documentation context for this might be.

**Comment by kinwa91** [comment]
@googleapis/yoshi-nodejs CI is not running for this PR. Does anyone know why?

**Comment by callmehiphop** [comment]
@kinwa91 in the past when we've seen it hang like this, usually some one just pushes a dummy commit to trigger a new build. As for why it happens, @alexander-fenster might have some insights?

**Comment by averikitsch** [comment]
The client library hasn't been published yet to npm, would this cause this issue?

**Comment by crwilcox** [comment]
@averikitsch , not being on npm shouldn't be an issue as the tests link to the version in the repo. If you run the following it should trigger a build.
```
git commit --allow-empty -m "Trigger Build"
git push REMOTENAME nodejs-tasks
```

**Comment by crwilcox** [comment]
I don't know why the CI for this isn't running. The only thing I can think about is that I run as a branch on this remote. @averikitsch has a fork.

**Comment by crwilcox** [comment]
Skipping CI. @averikitsch  ran these locally and they are fine.

**Comment by alexander-fenster** [comment]
@crwilcox That's a setting in CircleCI that can only be set manually, no API exists for that :(  It needs to be added to the playbook.

image

I changed it to On for tasks repo.

**Pull request merged**